Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,9 +48,75 @@ async function main(): Promise<void> {
         let pypircPath = utils.getPypircPath();
 
         // create new file. We do not merge existing files and always create a fresh file
-        fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
-        tl.setVariable("PYPIRC_PATH", pypircPath, false);
-        tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        if (!tl.getVariable("PYPIRC_PATH") || !tl.exist(tl.getVariable("PYPIRC_PATH"))) {
+            fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
+            tl.setVariable("PYPIRC_PATH", pypircPath, false);
+            tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        }
+        else {
+            pypircPath = tl.getVariable("PYPIRC_PATH");
+            const pypirc = fs.readFileSync(pypircPath, 'utf8');
+            let fileContent = ini.parse(pypirc);
+
+            let usedRepos = new Set<string>();
+
+            for (let connection in fileContent) {
+
+                const connectionObj: object = fileContent[connection];
+
+                if (!connectionObj.hasOwnProperty('repository')) {
+                    const authenticatedRepo = getNestedRepoProperty(connectionObj);
+
+                    if (authenticatedRepo === tl.loc("NoRepoFound")) {
```

Comment:
Is a localized string the right value to check for here?

Reply:
I think so because this is what getNestedRepoProperty returns if there is no repo listed in a .pypirc section

Reply:
see below comment on what the getNetstedRepo could return

Reply:
Changed this!